### ISSUE TITLE
1.114: PHP color highlight broken

### ISSUE THREAD

**Issue by m1ga** [pulsar-edit/pulsar#931]
### Thanks in advance for your bug report!

- [X] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [X] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [X] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [X] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [X] Have you checked you are on the latest release of Pulsar?

### What happened?

Opening a PHP file that starts with HTML code won't show any highlight colors with 1.114.

![Screenshot_20240216_111316](https://github.com/pulsar-edit/pulsar/assets/4334997/bee783a9-43ba-4315-83ce-3d7ff195febe)

If I move the `<?php ?>` block to the first line it will work.

### Pulsar version

1.114

### Which OS does this happen on?

🐧 Red Hat based (Fedora, Alma, RockyLinux, CentOS Stream, etc.)

### OS details

Fedora 39

### Which CPU architecture are you running this on?

x86_64/AMD64

### What steps are needed to reproduce this?

Create a new PHP file and add 
```html
<table>
  <thead>
    <tr></tr>
</thead>
</table>


<?php
echo 1;
?>
```

### Additional Information:

Works fine with 1.113, using experimental tree sitter

**Comment by savetheclocktower** [comment]
Thanks for the report. Is it a matter of it choosing the wrong grammar, or is the PHP grammar selected when the file is opened?

**Comment by m1ga** [comment]
PHP is selected in both tabs

**Comment by savetheclocktower** [comment]
Oof. I can't reproduce this with the exact code block you included.

Do you see any errors in the console?

If not, I’m at a loss here because most of the possible explanations I can think of for this behavior would involve partial highlighting; perhaps the HTML would be highlighted but the PHP wouldn't. At least if there's a console error it'd suggest that something was going wrong in the injection phase (where we figure out which parts are HTML and which parts are PHP).

**Comment by garrettboone** [comment]
I can reproduce with php selected as grammar. Seems to work fine at first if auto-detect is allowed to choose.

With `Auto Detect` showing as selected (which I think means no selection yet?) in grammar selection list and no other languages have a check mark next to them, though `PHP` is diplayed in the footer bar:
![image](https://github.com/pulsar-edit/pulsar/assets/47018474/460b3f41-f51d-4340-b464-983702d396a1)

With `PHP` selected with check mark showing on the list:
![image](https://github.com/pulsar-edit/pulsar/assets/47018474/e5d51438-9025-43d3-adf0-5b1790fee639)

The file is local to my Windows 10 machine. Project is remote. No errors occur in consolt when switching selection back and forth.

**Comment by garrettboone** [comment]
I cannot reproduce in 1.114.0 - here are the results:

With `Auto Detect`:
![image](https://github.com/pulsar-edit/pulsar/assets/47018474/facbb7d6-3d82-4e60-b6fa-6c9e996eb190)


With `PHP` selected:
![image](https://github.com/pulsar-edit/pulsar/assets/47018474/067362e5-acfb-4814-9b34-57ef3e861404)

These are with `Settings > Use Tree Sitter Parsers` unchecked. With checked, PHP is being selected from the list and checked.

**Comment by savetheclocktower** [comment]
My head hurts.

@m1ga, with @garrettboone's findings in mind, can you perhaps close your project window and relaunch it via the command line using

```shell
pulsar --clear-window-state [the path to your project]
```

to see if this is some strange artifact of window state?

And to be clear — @m1ga, your “Use Tree-sitter Parsers” setting is enabled, correct?

**Comment by m1ga** [comment]
It's strange. Today I reopened my file and test file and it is looking correct. I didn't change anything in my Pulsar settings or updated any package. The only thing I did between last time and today was working on other files and make git pushes.

![Screenshot_20240219_112828](https://github.com/pulsar-edit/pulsar/assets/4334997/44c108dd-49a9-41cb-baa4-5ffd273e0002)

so I guess I can close this one. It wasn't working on the day I've made the update of Pulsar
